crates/services/tx_status_manager/src/manager.rs

```diff
@@ -149,9 +181,11 @@ impl TxStatusManager {
 
     pub fn status(&self, tx_id: &TxId) -> Option<TransactionStatus> {
         self.data
-            .statuses
+            .prunable_statuses
             .get(tx_id)
-            .map(|(_, status)| status.clone())
+            .map(|(_, status)| status)
+            .or_else(|| self.data.non_prunable_statuses.get(tx_id))
```

Comment:
```suggestion
            .or_else(|| self.data.prunable_statuses.get(tx_id))
```

Reply:
Updated in https://github.com/FuelLabs/fuel-core/pull/2853/commits/5bf74053b5a7d39b5daaa7231bf9f99c9a392d4d